sdv/multi_table/base.py

```diff
@@ -318,6 +318,7 @@ def preprocess(self, data):
             synthesizer = self._table_synthesizers[table_name]
             self._assign_table_transformers(synthesizer, table_name, table_data)
             processed_data[table_name] = synthesizer._preprocess(table_data)
+            synthesizer._data_processor.reset_sampling()
```

Comment:
If this modification is intended exclusively for `HMA`, I believe overwriting it there would be the better approach.